src/test/scala/nl.knaw.dans.easy.deposit/StateManagerSpec.scala

```diff
@@ -0,0 +1,134 @@
+    draftPropsFile.writeText(
+      s"""state.label = SUBMITTED
+         |state.description = $message
+         |bag-store.bag-id = $uuid
+      """.stripMargin)
+    StateManager(draftDeposit, testDir / "does-not-exist").getStateInfo should matchPattern {
+      case Failure(e: PropertyNotFoundException) if e.getMessage ==
+        s"'state.label' not found in $testDir/does-not-exist/$uuid/deposit.properties" =>
+      // actually the submitted deposit does not exist
+    }
+  }
+
+  it should "change SUBMITTED to IN_PROGRESS" in {
+    val oldMessage = """The dataset is ready for processing"""
+    val newMessage = """The dataset is visible for you under your datasets in https://easy.dans.knaw.nl/ui"""
+    draftPropsFile.writeText(
+      s"""state.label = SUBMITTED
+         |state.description = $oldMessage
+         |bag-store.bag-id = $uuid
+      """.stripMargin)
+    submittedPropsFile.writeText(
+      s"""state.label = IN_REVIEW
+         |state.description = $newMessage
+      """.stripMargin)
+    StateManager(draftDeposit, submitBase).getStateInfo should matchPattern {
+      case Success(StateInfo(State.inProgress, `newMessage`)) =>
+    }
+  }
+
+  "setStateInfo" should "result in Success when transitioning from DRAFT to SUBMITTED" in {
+    draftPropsFile.writeText(
+      """state.label = DRAFT
+        |state.description = Deposit is open for changes.
+      """.stripMargin)
+    val triedUuid = StateManager(draftDeposit, submitBase)
+      .changeState(StateInfo(State.submitted, "rabarbera"))
+    val newUuid = triedUuid.getOrElse(fail(s"expecting Success but got $triedUuid"))
+    draftPropsFile.contentAsString should startWith(
+      s"""state.label = SUBMITTED
+         |state.description = rabarbera
+         |bag-store.bag-id = $newUuid
+         |""".stripMargin)
+  }
+
+  it should "result in Success when transitioning from REJECTED to DRAFT" in {
+    draftPropsFile.writeText(
+      s"""state.label = REJECTED
+         |state.description = Something's rotten on the state of ...
+         |bag-store.bag-id = $uuid
+         |""".stripMargin)
+    StateManager(draftDeposit, submitBase)
+      .changeState(StateInfo(State.draft, "rabarbera")) shouldBe Success(null)
+    draftPropsFile.contentAsString shouldBe
+      s"""state.label = DRAFT
+         |state.description = rabarbera
+         |""".stripMargin
+  }
+
+  it should "result in IllegalStateTransitionException when transitioning from DRAFT to ARCHIVED" in {
+    val props =
+      s"""state.label = DRAFT
+         |state.description = Something's rotten on the state of ...
+         |bag-store.bag-id = $uuid
+         |""".stripMargin
+    draftPropsFile.writeText(props)
+    StateManager(draftDeposit, submitBase)
+      .changeState(StateInfo(State.archived, "rabarbera")) should matchPattern {
+      case e: IllegalStateTransitionException if e.getMessage == "Cannot transition from DRAFT to ARCHIVED" =>
```

Comment:
@rvanheest Don't see the different between expectations and actual results. Otherwise this pull request is finished.